src/libsystemd/sd-device/test-sd-device-monitor.c

```diff
@@ -377,20 +378,25 @@ TEST(sd_device_monitor_receive) {
         ASSERT_OK(device_monitor_send(monitor_server, &sa, device));
 
         ASSERT_OK(fd = sd_device_monitor_get_fd(monitor_client));
+        ASSERT_OK(event = sd_device_monitor_get_events(monitor_client));
+        ASSERT_EQ(event, (int) EPOLLIN);
+        ASSERT_OK(sd_device_monitor_get_timeout(monitor_client, &timeout));
+        ASSERT_EQ(timeout, USEC_INFINITY);
+
         for (;;) {
-                r = fd_wait_for_event(fd, POLLIN, 10 * USEC_PER_SEC);
+                r = fd_wait_for_event(fd, event, 10 * USEC_PER_SEC);
```

Comment:
maybe change last arg to `MAX(10 * USEC_PER_SEC, timeout)`? it doesn't change behaviour, but looks more correct...